src/devices_models/devices/TwoTerminalDC_branches.jl

```diff
@@ -748,13 +697,599 @@ function add_constraints!(
+        bus_to = PSY.get_arc(d).to
+        bus_to_name = PSY.get_name(bus_to)
+
+        for t in get_time_steps(container)
+            constraint_inv_dc_volt[name, t] = JuMP.@constraint(
+                get_jump_model(container),
+                inv_dc_voltage_var[name, t] ==
+                (3 * inv_bridges / pi) * (
+                    sqrt(2) * (
+                        inv_tap_ratio *
+                        inv_ac_voltage_bus_var[bus_to_name, t] *
+                        cos(inv_extinction_angle_var[name, t])
+                    ) / inv_tap_setting_var[name, t] -
+                    dc_inv_com_reactance * dc_line_current_var[name, t]
+                )
+            )
+        end
+    end
+    return
+end
+
+function add_constraints!(
+    container::OptimizationContainer,
+    ::Type{HVDCRectifierOverlapAngleConstraint},
+    devices::IS.FlattenIteratorWrapper{T},
+    ::DeviceModel{T, <:HVDCTwoTerminalLCC},
+    ::NetworkModel{PM.ACPPowerModel},
+) where {T <: PSY.TwoTerminalLCCLine}
+    time_steps = get_time_steps(container)
+    names = PSY.get_name.(devices)
+    rect_ac_voltage_bus_var = get_variable(container, VoltageMagnitude(), PSY.ACBus)
+    rect_delay_angle_var = get_variable(container, HVDCRectifierDelayAngleVariable(), T)
+    rect_overlap_angle_var = get_variable(container, HVDCRectifierOverlapAngleVariable(), T)
+    rect_tap_setting_var = get_variable(container, HVDCRectifierTapSettingVariable(), T)
+    dc_line_current_var = get_variable(container, DCLineCurrentFlowVariable(), T)
+
+    constraint_rect_over_ang = add_constraints_container!(
+        container,
+        HVDCRectifierOverlapAngleConstraint(),
+        T,
+        names,
+        time_steps;
+    )
+
+    for d in devices
+        name = PSY.get_name(d)
+        dc_rect_com_reactance = PSY.get_rectifier_xc(d)
+        rect_tap_ratio = PSY.get_rectifier_transformer_ratio(d)
+        bus_from = PSY.get_arc(d).from
+        bus_from_name = PSY.get_name(bus_from)
+
+        for t in get_time_steps(container)
+            constraint_rect_over_ang[name, t] = JuMP.@constraint(
+                get_jump_model(container),
+                rect_overlap_angle_var[name, t] == (
+                    acos(
+                        cos(rect_delay_angle_var[name, t])
+                        -
+                        (
+                            (
+                                sqrt(2) * dc_rect_com_reactance *
+                                dc_line_current_var[name, t] *
+                                rect_tap_setting_var[name, t]
+                            )
+                            /
+                            (
+                                rect_tap_ratio *
+                                rect_ac_voltage_bus_var[bus_from_name, t]
+                            )
+                        ),
+                    )
+                    -
+                    rect_delay_angle_var[name, t]
```

Comment:
Can this be rewritten so it isn't so spread out? Somewhat hard to see what's going on (same below)